src/models/treemodel.h

```diff
@@ -281,6 +278,46 @@ class CostTreeModel : public TreeModel<decltype(Results::root), ModelImpl>
     Results m_results;
 };
 
+template<typename Results, typename ModelImpl>
+class CostProxy : public QSortFilterProxyModel
+{
+public:
+    explicit CostProxy(QObject* parent = nullptr)
+    {
+        setRecursiveFilteringEnabled(true);
+        setDynamicSortFilter(true);
+    }
+
+protected:
+    bool filterAcceptsRow(int source_row, const QModelIndex& parent) const override
+    {
+        if (parent.isValid())
+            return false;
+
+        const auto* model = static_cast<TreeModel<Results, ModelImpl>*>(sourceModel());
+
+        const auto* item = model->itemFromIndex(model->index(source_row, 0));
+
+        const auto needle = filterRegExp().pattern();
+
+        if (item->symbol.symbol.indexOf(needle) == -1 && item->symbol.binary.indexOf(needle) == -1) {
+            return false;
+        }
+
+        return true;
+    }
+
+    bool lessThan(const QModelIndex& source_left, const QModelIndex& source_right) const override
```

Comment:
this is wrong and should be removed again - you are always sorting by symbol here, which totally breaks sorting by costs, no?

generally, sorting is unrelated to filtering, so please remove this hunk from this patch and instead explicitly take the sort role in the ctor in addition to the symbol role, but just keep the default implementation for lessThan